test/tools/libtesteth/TestHelper.cpp

```diff
@@ -95,6 +95,15 @@ void mine(BlockHeader& _bi, SealEngineFace* _sealer, bool _verify)
 namespace test
 {
 
+void tryRunSingleTestFile(dev::test::TestSuite const& _suite)
+{
+	if (test::Options::get().singleTestFile.is_initialized())
+	{
+		boost::filesystem::path file(test::Options::get().singleTestFile.get());
+		_suite.runTestWithoutFiller(file);
+		exit(0);
```

Comment:
Why do you `exit` suddenly?  The function should usually just return when it finishes the task.

Maybe this function can return a boolean.

Reply:
because it should stop execution of other test cases in the suite

Reply:
Can you do something as if `--singletest 1234543212345` so that no test cases match?

Reply:
yes. it will scan the tests and if there are no test with such name - nothing will be executed.

Reply:
Can you instead of this have a function that calls either `_suite.runTestWithoutFiller(file);`  or `_suite.runAllTestsInFolder(casename);` depending on the result of `test::Options::get().singleTestFile.is_initialized()` ?
Will it help get rid of `exit` ?

Reply:
no. because we have many test cases called. but if singleTest is defined we have to stop the execution of all of them

Reply:
no. because we have many test cases called. but if singleTest is defined we have to stop the execution of all of them